### PR TITLE
Brought back One Click deploy

### LINKED ISSUE #378: Provide better Guidance on how to get the project working from clean install
The guidance on getting the solution working from a clean install is causing confusion and needs to be tidied up.

Fixes: #373 

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [x] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
Follow the instructions from the main readme.md

### PR DESCRIPTION
**Pull request by ross-p-smith**
## Purpose
Brought back the One Click Deploy and updated the documentation to reflect this

Closes #378

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Click the "Deploy to Azure" button on the Readme

### CODE REVIEW

File: infra/main.bicep [hunk 1]
```diff
@@ -82,7 +82,7 @@ param azureOpenAIModelVersion string = '0613'
   'openai_function'
   'langchain'
 ])
-param orchestrationStrategy string = 'langchain'
+param orchestrationStrategy string = 'openai_function'
```
Comment (cecheta) [inline infra/main.bicep:85]: How come?

**Comment by cecheta** [comment]
Is it safe to set this default to `code`, since`azd up` doesn't work with the value set to `container`?

https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/blob/7f6629ad5fde743ae8c22f206d39646dea6004eb/infra/main.bicepparam#L14

File: infra/main.bicep (hunk 1 above)
Comment (ross-p-smith) [reply infra/main.bicep:85]: The documentation states that we should be using `openai_function` if using 613 (which is the default) - so just aligning that.

**Comment by ross-p-smith** [comment]
@cecheta  - default from azd is now code

File: README.md [hunk 2]
```diff
@@ -129,32 +129,18 @@ The following are links to the pricing details for some of the resources:
 
 ### Deploy instructions
 
+You need to decide if you would like a quick demo like experience where everything is built for you, or whether you would like to debug and explore the code itself. The demo is available by clicking this button:
+
```
[the remark this answers is not in the document]
Comment (superhindupur) [reply README.md:133]: Can we also make it explicit that the container hosting model will always pull the latest container built from the main branch?

File: README.md [hunk 3]
```diff
@@ -129,32 +129,18 @@ The following are links to the pricing details for some of the resources:
...
+If you would like a developer experience, follow the [Local deployment instructions](./docs/LOCAL_DEPLOYMENT.md).
 
-**Notes:** the default auth type uses keys, if you want to switch to rbac, please run `azd env set AUTH_TYPE rbac`.
-
-
-#### Testing the deployment
+### Testing the deployment
 1. Navigate to the admin site, where you can upload documents. It will be located at:
     
     `https://{MY_RESOURCE_PREFIX}-website-admin.azurewebsites.net/`
```
[the remark this answers is not in the document]
Comment (ross-p-smith) [reply README.md]: Updated

File: infra/main.bicep [hunk 4]
```diff
@@ -147,22 +147,26 @@ param speechServiceName string = '${environmentName}-speechservice-${resourceTok
 
 param newGuidString string = newGuid()
 param searchTag string = 'chatwithyourdata-sa'
-param useKeyVault bool
+
+@description('Whether to use Key Vault to store secrets (best when using keys). If using RBAC, then please set this to false.')
```
[the remark this answers is not in the document]
Comment (ross-p-smith) [reply infra/main.bicep:151]: Done

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ross-p-smith) [reply README.md:135]: Now states

"The demo, which uses containers pre-built from the main branch is available by clicking this button:"

**Review by cecheta — DISMISSED** [review]

**Review by cecheta — APPROVED** [review]

**Pull request merged**